csrc/device_lower/analysis/tensor_memory.cpp

```diff
@@ -183,6 +188,180 @@ TMemAlllocationInfo computeTMemAlllocationInfo(Fusion* fusion) {
   return result;
 }
 
+// Get the TID Parallel types that we are interested in. We are not interested
+// in the parallel types that are not used in the kernel, and the ones that have
+// size 1. The order of the returned parallel types is from z to x.
+std::vector<ParallelType> getInterestedThreadParallelTypes(Fusion* fusion) {
+  const auto& pdim_map = GpuLower::current()->parallelDimensionMap();
+  std::vector<ParallelType> interested_tid_ptypes;
+  for (auto pt : {
```

Comment:
Would this work given `constexpr std::array<ParallelType, 3> kParallelTypeTIDs` in `csrc/type.h`?
```cpp
std::vector<ParallelType> interested_tid_ptypes;
interested_tid_ptypes.reserve(reverse_pt_order.size());
std::copy_if(kParallelTypeBIDs.rbegin(), kParallelTypeBIDs.rend(), std::back_inserter(interested_tid_ptypes), [](ParallelType pt) {
    Val* size = pdim_map.getRaw(pt);
    if (size == nullptr) {
      return false;
    }
    Val* size_is_one =
        simplifyExpr(SimplifyingIrBuilder::eqExpr(size, fusion->oneVal()));
    if (size_is_one->isTrue()) {
      return false;
    }
    return true;
  }
});
```

Reply:
Changed to
```c++
for (ParallelType pt : std::views::reverse(nontrivial_tid_ptypes))
```